ROS2 CI Error

https://github.com/openrr/openrr/actions/runs/3845220676/jobs/6549115323

```
Running tests/test_navigation.rs (/__w/openrr/openrr/target/debug/deps/test_navigation-8dca5a05a6407a25)

running 3 tests

>>> [rcutils|error_handling.c:108] rcutils_set_error_state()
This error state is being overwritten:

  'string_map already initialized, at ./src/string_map.c:57'

with this new error message:

  'Failed to initialize map for logger severities [string_map already initialized, at ./src/string_map.c:57]. Severities will not be configurable., at ./src/logging.c:286'

rcutils_reset_error() should be called after error handling to avoid this.
<<<
[rcutils|./src/rcl/init.c:71] error initializing logging: Failed to initialize map for logger severities [string_map already initialized, at ./src/string_map.c:57]. Severities will not be configurable., at ./src/logging.c:286
[WARN] [1672908962.879528333] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
test test_nav_cancel ... ok
test test_nav_timeout ... ok
test test_nav ... ok

test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 4.17s

error: test failed, to rerun pass `--test test_navigation`

Caused by:
  process didn't exit successfully: `/__w/openrr/openrr/target/debug/deps/test_navigation-8dca5a05a6[40](https://github.com/openrr/openrr/actions/runs/3845220676/jobs/6549115323#step:9:41)7a25` (signal: 11, SIGSEGV: invalid memory reference)
Error: Process completed with exit code 101.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -148,6 +148,7 @@ jobs:
       image: docker://ros:${{ matrix.distro }}
     env:
       HOME: /root
+      RUST_TEST_THREADS: 1
     steps:
       - uses: actions/checkout@v3
       - run: |

--- a/arci-ros2/src/cmd_vel_move_base.rs
+++ b/arci-ros2/src/cmd_vel_move_base.rs
@@ -1,12 +1,13 @@
 use arci::*;
+use parking_lot::Mutex;
 use r2r::geometry_msgs::msg::Twist;
 use serde::{Deserialize, Serialize};
 
 /// Implement arci::MoveBase for ROS2
 pub struct Ros2CmdVelMoveBase {
-    vel_publisher: r2r::Publisher<Twist>,
+    vel_publisher: Mutex<r2r::Publisher<Twist>>,
     // keep not to be dropped
-    _node: r2r::Node,
+    _node: Mutex<r2r::Node>,
 }
 
 impl Ros2CmdVelMoveBase {
@@ -22,16 +23,14 @@ impl Ros2CmdVelMoveBase {
     #[track_caller]
     pub fn from_node(mut node: r2r::Node, cmd_topic_name: &str) -> Self {
         Self {
-            vel_publisher: node
-                .create_publisher(cmd_topic_name, r2r::QosProfile::default())
-                .unwrap(),
-            _node: node,
+            vel_publisher: Mutex::new(
+                node.create_publisher(cmd_topic_name, r2r::QosProfile::default())
+                    .unwrap(),
+            ),
+            _node: Mutex::new(node),
         }
     }
 }
-
-// TODO:
-unsafe impl Sync for Ros2CmdVelMoveBase {}
 
 impl MoveBase for Ros2CmdVelMoveBase {
     fn send_velocity(&self, velocity: &BaseVelocity) -> Result<(), Error> {
@@ -40,6 +39,7 @@ impl MoveBase for Ros2CmdVelMoveBase {
         twist_msg.linear.y = velocity.y;
         twist_msg.angular.z = velocity.theta;
         self.vel_publisher
+            .lock()
             .publish(&twist_msg)
             .map_err(|e| arci::Error::Connection {
                 message: format!("r2r publish error: {e:?}"),

--- a/arci-ros2/src/navigation.rs
+++ b/arci-ros2/src/navigation.rs
@@ -26,9 +26,6 @@ pub struct Ros2Navigation {
     current_goal: Arc<Mutex<Option<r2r::ActionClientGoal<NavigateToPose::Action>>>>,
 }
 
-// TODO:
-unsafe impl Sync for Ros2Navigation {}
-
 impl Ros2Navigation {
     /// Creates a new `Ros2Navigation` from ROS2 context and name of action.
     #[track_caller]

--- a/arci-ros2/src/ros2_control.rs
+++ b/arci-ros2/src/ros2_control.rs
@@ -28,9 +28,6 @@ pub struct Ros2ControlClient {
     joint_names: Vec<String>,
 }
 
-// TODO:
-unsafe impl Sync for Ros2ControlClient {}
-
 impl Ros2ControlClient {
     /// Creates a new `Ros2ControlClient` from ROS2 context and names of action.
     #[track_caller]

